TF_Settings_Web/src/Pages/Visuals/VisualsScreen.tsx

```diff
@@ -0,0 +1,273 @@
+import styles from './Visuals.module.scss';
+
+import classNames from 'classnames/bind';
+import React, { useState, useEffect, useRef, useReducer, useMemo } from 'react';
+import tinycolor, { ColorFormats } from 'tinycolor2';
+
+import { readVisualsConfig, isDesktop, writeVisualsConfig } from '@/TauriUtils';
+import { useIsLandscape } from '@/customHooks';
+
+import {
+    BlackTextBg,
+    BlackTextBgPreview,
+    GradientBg,
+    GradientBgPreview,
+    MountainBg,
+    MountainBgPreview,
+    WhiteTextBg,
+    WhiteTextBgPreview,
+} from '@/Images';
+
+import {
+    DocsLink,
+    FileInput,
+    LabelledToggleSwitch,
+    OutlinedTextButton,
+    RadioGroup,
+    RadioLine,
+    TextSlider,
+} from '@/Components';
+
+import ColorPicker from './ColorPicker';
+import { defaultCtiVisualsConfig } from './VisualsUtils';
+import {
+    CursorPreset,
+    cursorStyles,
+    VisualsConfig,
+    defaultVisualsConfig,
+    CursorStyle,
+    presets,
+    defaultCursorVisualsConfig,
+} from './VisualsUtils';
+
+const classes = classNames.bind(styles);
+
+const styleOptions = Object.keys(cursorStyles);
+const bgImages = [GradientBg, WhiteTextBg, BlackTextBg, MountainBg];
+const bgPreviewImages = [GradientBgPreview, WhiteTextBgPreview, BlackTextBgPreview, MountainBgPreview];
+const closeCtiOptions = ['Users Hand Present', 'User Performs Interaction'];
+
+const reducer = (state: VisualsConfig, payload: { content: Partial<VisualsConfig>; writeOutConfig: boolean }) => {
+    const newState: VisualsConfig = { ...state, ...payload.content };
+    if (payload.writeOutConfig) {
+        writeVisualsConfig(newState).catch((err) => console.error(err));
+    }
+    return newState;
+};
+
+const VisualsScreen: React.FC = () => {
+    const isLandscape = useIsLandscape();
+    const cursorSection = useRef<HTMLDivElement>(null);
+
+    const [state, dispatch] = useReducer(reducer, defaultVisualsConfig);
+    const writeOutConfig = () => dispatch({ content: {}, writeOutConfig: true });
+    const [hasReadConfig, setHasReadConfig] = useState<boolean>(false);
+
+    const [currentPreviewBgIndex, setCurrentPreviewBgIndex] = useState<number>(0);
+
+    const updateCursorPreview = (cursorStyle: CursorStyle) => {
+        const section = cursorSection.current?.style;
+        if (!section) return;
+
+        section.setProperty('--center-fill', cursorStyle[0]);
+        section.setProperty('--outer-fill', cursorStyle[1]);
+        section.setProperty('--center-border', cursorStyle[2]);
+    };
+
+    const currentPreset = useMemo((): CursorPreset => {
+        return presets[state.activeCursorPreset];
+    }, [state.activeCursorPreset]);
```

Comment:
:shrug: Could be one line without the `return`

Reply:
Puts it all on one line which feels less readable